fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java

```diff
@@ -257,7 +260,22 @@ public void delete() {
             final Iterator<Property> inboundProperties = Iterators.concat(references, weakReferences);
 
             while (inboundProperties.hasNext()) {
-                inboundProperties.next().remove();
+                final Property prop = inboundProperties.next();
+                final List<Value> newVals = new ArrayList<Value>();
+                final Iterator<Value> propIt = JcrPropertyFunctions.property2values.apply(prop);
+                while (propIt.hasNext()) {
```

Comment:
Ok, that is interesting, but I'm not sure that using a Predicate doesn't make the code more confusing.
Also the **property2values** function returns an UnmodifiableIterator, so the **removeIf** fails to modify it.
I put this attempt in a [separate branch](https://github.com/whikloj/fcrepo4/blob/FCREPO-1362-2/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java#L262-L283) so you can see what I've done, incase I missed something.